### PR TITLE
Support for building github desktop offline

### PR DESCRIPTION
**Pull request by Lunarequest**
Assuming a offline yarn mirror has been setup  like [here](https://classic.yarnpkg.com/blog/2016/11/24/offline-mirror/). By setting the env var `OFFLINE` to 1 you can  build with offline. This is useful for situations where internet access is not possible, such as in a flatpak or in the Fedora build infrastructure.

### CODE REVIEW

File: script/post-install.ts [hunk 1]
```diff
@@ -27,36 +27,53 @@ function findYarnVersion(callback: (path: string) => void) {
...
   if (result.status !== 0) {
     process.exit(result.status || 1)
   }
-
-  result = spawnSync('node', [path, 'compile:tslint'], options)
-
+  if (isOffline()) {
+    result = spawnSync('node', [path, '--offline', 'compile:tslint'], options)
+  } else {
+    result = spawnSync('node', [path, '--offline', 'compile:tslint'], options)
```
Comment (shiftkey) [inline script/post-install.ts]: :art: the `--offline` should be omitted from here:

```suggestion
    result = spawnSync('node', [path, 'compile:tslint'], options)
```

**Comment by Lunarequest** [comment]
I have no idea why the windows ci broke, it seems like some sort of network issue?

**Comment by Lunarequest** [comment]
@shiftkey would you mind reviewing this pr in its altered state?

**Comment by shiftkey** [comment]
@advaithm I've held off on review because I wanted to suggest a different approach, and I've not had time to prove it out. Instead of this pattern:

```ts
  let result
   if (isOffline()) {
     result = spawnSync(
       'node',
       [path, '--offline', '--cwd', 'app', 'install', '--force'],
       options
     )
   } else {
     result = spawnSync(
       'node',
       [path, '--cwd', 'app', 'install', '--force'],
       options
     )
   }
```

I wanted to see if we could avoid duplicating the arguments and instead insert the relevant offline parameter at the appropriate place:

```ts
  const installArgs = [path, '--cwd', 'app', 'install', '--force']
  if (isOffline()) {
    installArgs.splice(1, 0, '--offline')
  }

  let result = spawnSync('node', installArgs, options)
```

Tomorrow I should have some more bandwidth, but hopefully that helps with some feedback in the meantime...

**Pull request closed without merging**